[Utils] [Chat Formatting] New rows to embeds util

FixedThink opened the pull request:
### Type

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature

### Description of the changes
- Added `rows_to_embeds()` util in `chat_formatting`.
- Added two unit tests for the util.

Resolves #4600.

FixedThink committed:
Add rows_to_embeds util

added rows_to_embeds util (in chat_formatting)
added two unit tests

FixedThink committed:
Fix docstring formatting of rows_to_embeds

Drapersniper commented on redbot/core/utils/chat_formatting.py:
```diff
@@ -283,6 +284,103 @@ def pagify(
+    ------
+    ValueError
+        When either ``embed_max_fields`` or ``field_max_rows`` is not positive.
+    """
```
We should cap this to the api limited value too

Drapersniper commented on redbot/core/utils/chat_formatting.py:
```diff
@@ -283,6 +284,103 @@ def pagify(
+    # Initialise for nested loop.
+    embed_list: List[discord.Embed] = []
+    start = 0
+    f = 0
```
This doesn't appear to be used anywhere

FixedThink replied:
- `embed_list` is used in L380 (to append)
- `start` is used several times (lines 371, 373m 376)
- `f` is incremented in L379

Drapersniper replied:
I meant specifically f, it's increments but I don't see it being used anywhere other then in the line which it gets incremented

Drapersniper commented on redbot/core/utils/chat_formatting.py:
```diff
@@ -283,6 +284,103 @@ def pagify(
+        The strings that form a row in the embed
+    base_embed : `discord.Embed`
+        The embed to use as a base for the list of embeds
+    embed_max_fields : `int`, optional
```
The optional typing here should be Optiona[type] just to follow the existing format, this also applies to the other instances in the docstring

Drapersniper reviewed (changes requested):
Just some things I've noticed at a glance ... I'm not a fan of the util name but can't think of an alternative either

FixedThink committed:
Rrename function, update docstring

- Renamed function from `rows_to_embeds` to `embeds_from_rows`
- Changed "type, optional" to "Optional[type]".
- Add embed field limit check (max = 25).

FixedThink commented:
Updated the files based on the comments above. I renamed the function to `embeds_from_rows`, maybe that sounds a bit better? At least the first word now makes clear what the function returns.

bobloy commented on redbot/core/utils/chat_formatting.py:
```diff
@@ -284,7 +284,7 @@ def pagify(
 
 
-def rows_to_embeds(
+def embeds_from_rows(
```
```suggestion
def embedify_list(
```
Alternative name option. Follows the naming scheme of `pagify`

bobloy commented on tests/core/test_utils.py:
```diff
@@ -203,7 +203,7 @@ def test_embed_list():
     rows = list(str(i) for i in range(11))
 
-    embed_list = chat_formatting.rows_to_embeds(
+    embed_list = chat_formatting.embeds_from_rows(
```
```suggestion
    embed_list = chat_formatting.embedify_list(
```

bobloy commented on tests/core/test_utils.py:
```diff
@@ -223,7 +223,7 @@ def test_embed_list_2():
     rows = list(str(i) for i in range(11))
 
-    embed_list = chat_formatting.rows_to_embeds(
+    embed_list = chat_formatting.embeds_from_rows(
```
```suggestion
    embed_list = chat_formatting.embedify_list(
```

bobloy reviewed (commented):
Alternative naming suggestion: `embedify_list`

Flame442 committed:
Merge branch 'V3/develop' into rows-embeds-util

Jackenmen commented:
Sorry that this has been open for so long without any feedback...
While this may have some use, I don't really​ want to commit to a design without a bunch of concrete usage examples *with* code snippets and screenshots. I'll close this and the corresponding issue because I don't think it makes to keep this in limbo forever... *Maybe* this would be useful, but until aforementioned concrete usage examples are provided, this wasn't going to progress anyway...
